lisa/sut_orchestrator/azure/features.py

```diff
@@ -2151,6 +2162,135 @@ def on_before_deployment(cls, *args: Any, **kwargs: Any) -> None:
                     )
 
 
+class Availability(AzureFeatureMixin, features.Availability):
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        super()._initialize(*args, **kwargs)
+        self._initialize_information(self._node)
+
+    @classmethod
+    def settings_type(cls) -> Type[schema.FeatureSettings]:
+        return AvailabilitySettings
+
+    @classmethod
+    def create_setting(
+        cls, *args: Any, **kwargs: Any
+    ) -> Optional[schema.FeatureSettings]:
+        raw_capabilities: Any = kwargs.get("raw_capabilities")
+        availability_settings: AvailabilitySettings = AvailabilitySettings()
+        availability_settings.availability_type = search_space.SetSpace(
+            True,
+            [
+                AvailabilityType.NoRedundancy,
+                AvailabilityType.AvailabilitySet,
+            ],
+        )
+
+        availability_zones = raw_capabilities.get("availability_zones", None)
+        if availability_zones:
+            availability_settings.availability_type.add(
+                AvailabilityType.AvailabilityZone
+            )
+            availability_settings.availability_zones = search_space.SetSpace(
+                is_allow_set=True, items=availability_zones
+            )
+
+        return availability_settings
+
+    @classmethod
+    def on_before_deployment(cls, *args: Any, **kwargs: Any) -> None:
+        environment = cast(Environment, kwargs.get("environment"))
+        arm_parameters = cast(AzureArmParameter, kwargs.get("arm_parameters"))
+        params = arm_parameters.availability_options
+
+        is_ultra_disk = any(
+            [
+                node.data_disk_type == _disk_type_mapping[schema.DiskType.UltraSSDLRS]
```

Comment:
It's better to apply checks on node or node space, instead of arm_parameters. Arm_parameters is more like an output.

Reply:
Environment.nodes was empty. What other variable contains maximize_capability?

Reply:
try `environment.runbook.nodes_requirement`

Reply:
What is the difference between
`environment.runbook.nodes_requirement[0].disk` and
`environment.capability.nodes[0].disk`?
Should I use one vs. the other?

Reply:
environment.runbook.nodes_requirement is almost here in Azure, but nodes are created later.

Reply:
Sorry, I mixed it with places. In `on_before_deployment`, the nodes should be ready to use now. So use Node is better than node_requirements, in case it breaks in a combined ready/azure environment. The combined env is supported, but not used in real scenarios.